yarn-project/types/src/tx_hash.ts

```diff
@@ -9,12 +9,19 @@ export class TxHash {
    */
   public static SIZE = 32;
 
+  /**
+   * The underlying hash data.
+   */
+  public readonly buffer: Buffer;
+
   constructor(
     /**
      * The buffer containing the hash.
      */
-    public readonly buffer: Buffer,
-  ) {}
+    private input: Buffer,
+  ) {
+    this.buffer = Buffer.concat([Buffer.alloc(TxHash.SIZE - input.length), input]);
```

Comment:
I'd rather have a static `fromBuffer28` method that checks the input is 224 bits and does the padding, and have this constructor validate that the input is 32 bytes, so we can note early if we're accidentally passing a buffer that we shouldn't have.